### ISSUE TITLE
Webview doc/sample feedback

### ISSUE THREAD

**Issue by octref** [Microsoft/vscode#48705]
Testing #48453

1. Should distinguish watch/compile scripts (https://github.com/Microsoft/vscode-extension-samples/pull/63)

2. The guide follows a "building-up" approach, so I think it should link to the yeoman generator for generating a skeleton project first. By only offering the full app, some people who get directly linked to this page might not be able to code along reading the docs.

3. In the sample it's `catCoding.start` whereas in the doc it's `catCoding.newCat`. This makes it harder to copy-paste the code and be able to run it immediately.

~~Still Editing this issue. Not done yet.~~

All through with the doc.

**Comment by mjbvz** [comment]
Fixed 1 and 3.

I'm not sure about 2. This is an advanced api feature so the target audience is really small. If we create a simple webview boilerplate, we are sending the message that this API is something that we think many extension authors should use (which is not the case)

4 is covered in the docs. The section on `retainContextWhileHidden` explains that this option has high memory overhead. We cannot stop people for using this option just because it is convenient, but now we have documented what the cost of this choice is

For 5, there's already basic section on how to open the developer tools in the docs. I'll add a note on reloading too
